### PR TITLE
implements Anansi driver and daemon

### PR DESCRIPTION
**Pull request by Zerophase**
driver and daemon support Anansi

### CODE REVIEW

File: driver/razerkbd_driver.h [hunk 1]
```diff
@@ -74,6 +74,9 @@
     #define USB_DEVICE_ID_RAZER_BLADE_QHD 0x020F
 #endif
 
+#ifndef USB_DEVICE_ID_RAZER_ANANSI
+    #define USB_DEVICE_ID_RAZER_ANANSI 0x010F
```
Comment (terricain) [inline driver/razerkbd_driver.h]: Put it in order of VID :wink: (and sort out any unordered ones too)

File: driver/razeranansicommon.c [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#include "razeranansicommon.h"
+
+struct razer_report razer_anansi_enable_macro_flash()
+{
```
Comment (terricain) [inline driver/razeranansicommon.c]: This command technically isnt specific to the anansi... it just isnt used elsewhere. (its listed in the fw spec I have, it just listed without any description ha)

Replace with the following and put that in razerchromacommon.c, and the appropriate entry into razerchromacommon.h, then we have 1 less file.
```c
struct razer_report razer_chroma_standard_set_led_blinking_params(unsigned char variable_storage, unsigned char led_id)
{
	struct razer_report report = get_razer_report(0x03, 0x04, 0x04);
	report.arguments[0] = variable_storage; // Effect ID
	report.arguments[1] = led_id; // LED ID
	report.arguments[2] = 0x05;
	report.arguments[3] = 0x05;
	
	return report;
}
```


Where variable_storage would be one of the defines NOSTORE or VARSTORE and led_id should be MACRO_LED,

I have a feeling arg 2,3 could possibly be on and off timing.

**Review by terricain — CHANGES_REQUESTED** [review]
Github aint letting me comment on razerkbd.c

Normally in `razer_attr_write_mode_none` we dont turn off game and macro led, we leave that to the daemon (though technically the daemon doesn't do that either ha (but it could...))

Line 642 `+ * Write device file "mode_reactive"?`
Is the `?` needed?

Line 988-993 seems like that shouldn't be there.
```c
+        case USB_DEVICE_ID_RAZER_BLADE_STEALTH:
+        case USB_DEVICE_ID_RAZER_BLADE_STEALTH_LATE_2016:
+            report = razer_chroma_misc_get_blade_brightness();
+            break;
+        
+  
```

Line 1000 is not needed as would fall though to default anyway.
```c
+        case USB_DEVICE_ID_RAZER_ANANSI:
```

File: driver/razeranansicommon.c (hunk 2 above)
Comment (terricain) [reply driver/razeranansicommon.c]: Obviously test it once its swapped out for this :wink:

**Comment by terricain** [comment]
Other than those changes, brilliant job mate. You tested the daemon right, either directly or via the python library or an app?

File: driver/razeranansicommon.c (hunk 2 above)
Comment (Zerophase) [reply driver/razeranansicommon.c]: I'm trying to test by setting up a blinking device file, but for some reason it won't show up when I reload the driver. I set up a `DEVICE_ATTR`, and made sure to call `CREATE_DEVICE_FILE` and `device_remove_device`

Comment (Zerophase) [reply driver/razeranansicommon.c]: This works for making the macro led blink as well. I tried changing argument 2, and the state of the macro led didn't change.

```c
struct razer_report report = get_razer_report(0x03, 0x04, 0x04);
report.arguments[0] = variable_storage;
report.arguments[1] = led_id;
report.arguments[2] = 0x05;
report.arguments[3] = 0x01;

```

**Comment by Zerophase** [comment]
I believe everything meets your needs now.

**Comment by terricain** [comment]
All looks good, will have a proper look tonight but should be good to merge. Just to be sure, you tested it all right?

**Comment by Zerophase** [comment]
Yeah, I tested everything manually.

**Review by terricain — APPROVED** [review]

**Comment by terricain** [comment]
:). If your up for it, see what [changing](https://github.com/Zerophase/razer-drivers/blob/b1357787917ec65821766fc7a753d15ebc232af9/driver/razerchromacommon.c#L91) the `0x05 0x05` to `0x02 0x02` does. I have a feeling it'll make it flash faster.

**Comment by Zerophase** [comment]
It's about the same with either setting. Don't think it does anything for this keyboard at least.

**Comment by terricain** [comment]
Ahh np

**Comment by z3ntu** [comment]
@Zerophase Can you please try the newest version of the driver if eg brightness still works? We have a report in #347 that says otherwise.

**Comment by Zerophase** [comment]
Sure, just give me a sec.

**Comment by Zerophase** [comment]
Yeah, it's not working currently. Going to have to dig around over the next couple days to see where the issue is. Looking at the old and current code I don't see any differences that could have broken it.

**Pull request merged**